c_emulator/rvfi_dii.h

```diff
@@ -23,7 +23,222 @@ class rvfi_handler {
 private:
   void get_and_send_packet(packet_reader_fn reader, bool config_print);
 
+  void send_packet_raw(const void *data, size_t len, bool config_print);
   unsigned trace_version = 1;
   int dii_port = -1;
   int dii_sock = -1;
 };
+
+// ****************************************************************************
+
+typedef struct {
```

Comment:
Since we are using C++ there is no need for typedef struct, we can just use a named struct.
```suggestion
struct RVFI_DII_Execution_Packet_InstMetadata {
```